packages/dev/core/src/Engines/engine.ts

```diff
@@ -1893,27 +1891,28 @@ export class Engine extends ThinEngine {
         this.disableVR();
 
         // Events
-        if (IsWindowObjectExist()) {
-            window.removeEventListener("blur", this._onBlur);
-            window.removeEventListener("focus", this._onFocus);
-
-            if (this._renderingCanvas) {
-                this._renderingCanvas.removeEventListener("focus", this._onCanvasFocus);
-                this._renderingCanvas.removeEventListener("blur", this._onCanvasBlur);
-                this._renderingCanvas.removeEventListener("pointerout", this._onCanvasPointerOut);
-                this._renderingCanvas.removeEventListener("contextmenu", this._onCanvasContextMenu);
-            }
+        const hostWindow = this.getHostWindow(); // it calls IsWindowObjectExist()
+        if (hostWindow && typeof hostWindow.removeEventListener === 'function') {
+            hostWindow.removeEventListener("blur", this._onBlur);
+            hostWindow.removeEventListener("focus", this._onFocus);
+        }
 
-            if (IsDocumentAvailable()) {
-                document.removeEventListener("fullscreenchange", this._onFullscreenChange);
-                document.removeEventListener("mozfullscreenchange", this._onFullscreenChange);
-                document.removeEventListener("webkitfullscreenchange", this._onFullscreenChange);
-                document.removeEventListener("msfullscreenchange", this._onFullscreenChange);
-                document.removeEventListener("pointerlockchange", this._onPointerLockChange);
-                document.removeEventListener("mspointerlockchange", this._onPointerLockChange);
-                document.removeEventListener("mozpointerlockchange", this._onPointerLockChange);
-                document.removeEventListener("webkitpointerlockchange", this._onPointerLockChange);
-            }
+        if (this._renderingCanvas) {

```

Comment:
I removed this `if` from within the `if (IsWindowObjectExist()) ` because they don't seem related and we should try to clean as many things as possible, but I might be wrong, you tell me :)

Reply:
it feels like a correct assumption :-)